### ISSUE TITLE
After several iterations gradient norm and loss becomes nan

### ISSUE THREAD

**Issue by neurosynapse** [microsoft/Swin-Transformer#287]
Hello,

It would be nice if you could help me to solve this issue. I have been trying to train the swin transformer model (swin_base_patch4_window7_224) on imagenet dataset with 100 classes (I exchanged the mlp head from 1000 to 100 dimension output). However, after some iterations the gradient norm and loss become nan. I have tried several lr and gradient clip values but the issue persists. 

![image](https://user-images.githubusercontent.com/47304445/199077175-e2f866fa-92ce-43fd-a2a6-d70393d6eb90.png)

Best regards,
Roberts

**Comment by byronyi** [comment]
Same issue here, except I am using all default config and hyper parameters for Swin-B from scratch on ImageNet-1k. Gradient norm explodes quickly after several epochs.

**Comment by rajeevgl01** [comment]
This issue is due to AMP using the torch.float16 dtype by default. Use torch.bfloat16 instead.
